server/src/main/java/io/crate/expression/operator/all/AllEqOperator.java

```diff
@@ -0,0 +1,126 @@
+import io.crate.expression.symbol.Symbol;
+import io.crate.lucene.LuceneQueryBuilder;
+import io.crate.metadata.FunctionType;
+import io.crate.metadata.Reference;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.sql.tree.ComparisonExpression;
+import io.crate.types.TypeSignature;
+
+public class AllEqOperator extends AllOperator {
+
+    public static String NAME = OPERATOR_PREFIX + ComparisonExpression.Type.EQUAL.getValue();
+    public static Signature SIGNATURE = Signature.builder(AllEqOperator.NAME, FunctionType.SCALAR)
+            .argumentTypes(TypeSignature.parse("E"), TypeSignature.parse("array(E)"))
+            .returnType(Operator.RETURN_TYPE.getTypeSignature())
+            .typeVariableConstraints(typeVariable("E"))
+            .features(Feature.DETERMINISTIC, Feature.STRICTNULL)
+            .build();
+
+    public AllEqOperator(Signature signature, BoundSignature boundSignature, IntPredicate cmp) {
+        super(signature, boundSignature, cmp);
+    }
+
+    public static Query refMatchesAllArrayLiteral(Reference probe, List<?> values, LuceneQueryBuilder.Context context) {
+        // col = ALL ([1,2,3]) --> col=1 and col=2 and col=3
+        LinkedHashSet<?> uniqueValues = new LinkedHashSet<>(values);
+        if (uniqueValues.contains(null)) {
+            return new MatchNoDocsQuery("Cannot match nulls");
+        }
+        if (uniqueValues.size() > 1) {
+            // `col=x and col=y` evaluates to `false` unless `x == y`
+            return new MatchNoDocsQuery("A single value cannot match more than one unique values");
+        }
+        String columnName = probe.storageIdent();
+        BooleanQuery.Builder builder = new BooleanQuery.Builder();
+        var value = uniqueValues.getFirst();
+        var fromPrimitive = EqOperator.fromPrimitive(
+            probe.valueType(),
+            columnName,
+            value,
+            probe.hasDocValues(),
+            probe.indexType());
+        if (fromPrimitive == null) {
+            return null;
+        }
+        builder.add(fromPrimitive, BooleanClause.Occur.MUST);
+        Query exists = IsNullPredicate.refExistsQuery(probe, context, false);
+        return builder.add(exists, BooleanClause.Occur.FILTER).build();
+    }
+
+    public static Query literalMatchesAllArrayRef(Literal<?> probe, Reference candidates, LuceneQueryBuilder.Context context) {
+        // 1 = ALL(array_col)
+        // --> 1 = array_col[1] and 1 = array_col[2] and 1 = array_col[3]
+        // --> not(1 != array_col[1] or 1 != array_col[2] or 1 != array_col[3])
+        // --> not(1 != ANY(array_col))
+        return new BooleanQuery.Builder()
+            .add(Queries.not(AnyNeqOperator.literalMatchesAnyArrayRef(probe, candidates)), BooleanClause.Occur.MUST)
+            .add(IsNullPredicate.refExistsQuery(candidates, context, false), BooleanClause.Occur.FILTER)
+            .build();
+    }
```

Comment:
<details open>
<summary>Benchmark results using specs/ignore3vl.toml</summary>

```
# Results (server side duration in ms)
V1: 5.9.0-ff540daa73a88db7d4392bde3c856a804f9cb27c
V2: 5.9.0-1022bc5d753eaa6ba5c86e10c83f3ecb3879bc50

Q: select count(*) from t where not 101 = all(a)
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      228.610 ±  250.696 |    184.668 |    190.199 |    195.310 |   2413.285 |
|   V2    |        7.980 ±   21.127 |      3.908 |      4.692 |      6.607 |    273.468 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               - 186.51%                           - 190.37%   
There is a 100.00% probability that the observed difference is not random, and the best estimate of that difference is 186.51%
The test has statistical significance

Q: select count(*) from t where not 50 = all(a)
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      203.290 ±  134.021 |    183.514 |    188.500 |    193.510 |   1704.565 |
|   V2    |        5.882 ±   15.062 |      3.968 |      4.615 |      5.098 |    262.136 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               - 188.75%                           - 190.44%   
There is a 100.00% probability that the observed difference is not random, and the best estimate of that difference is 188.75%
The test has statistical significance
```
</details>

Reply:
Imcomplete implementation see https://github.com/crate/crate/pull/16502#discussion_r1736946564 for an update.